### PR TITLE
feat: add Koi chain Testnet

### PR DESCRIPTION
**Pull request by guy93354**
### Description
The name of the network was chosen according to official documentation: https://docs.darwinia.network/build/getting-started/networks/koi/
It differs from the name on chainlist.org, but I assume the correct name is the one from the official docs.

`interchainGasPaymaster` set to "0x0000000000000000000000000000000000000000" as I couldn't find anything,
`merkleTreeHook` address added,
logo.svg ran in https://jakearchibald.github.io/svgomg/, but I don't know what to remove, so I hit "reset all". 
Please let me know what should be modified and I'll do it correctly for the next PR.

Add Koi chain Testnet

### Backward compatibility

Yes

### Testing

Yes

### CODE REVIEW

File: .changeset/seven-ducks-kiss.md
Comment (jmrossy) [inline .changeset/seven-ducks-kiss.md:1]: Did you mean to include this? It seems unrelated.

File: chains/koitestnet/logo.svg
Comment (jmrossy) [inline chains/koitestnet/logo.svg:1]: Please use a roughly square logo image. You can cut out the paths for the K, O, and I characters and then use these online tools to crop + minify the image:
https://svgcrop.com/
https://jakearchibald.github.io/svgomg/

File: chains/koitestnet/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  - apiUrl: https://koi-scan.darwinia.network/api
+    family: blockscout
+    name: Koi Explorer
+    url: https://koi-scan.darwinia.network/
+chainId: 701
+deployer:
+  name: guy93354
+  url: https://github.com/guy93354
+chainId: 701
+displayName: Koi chain Testnet
```
Comment (jmrossy) [inline chains/koitestnet/metadata.yaml:12]:
```suggestion
displayName: Koi Testnet
```

File: .changeset/seven-ducks-kiss.md
Comment (guy93354) [reply .changeset/seven-ducks-kiss.md:1]: I might have missclicked, you can ignore it... sorry.

**Review by guy93354 — COMMENTED** [review]
I modified what you requested, please let me know if it's not what you wanted.
Thank you.

File: .changeset/orange-mails-ring.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@hyperlane-xyz/registry": patch
```
Comment (jmrossy) [inline .changeset/orange-mails-ring.md:2]:
```suggestion
"@hyperlane-xyz/registry": minor
```

File: chains/koitestnet/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# yaml-language-server: $schema=../schema.json
+blockExplorers:
+  - apiUrl: https://koi-scan.darwinia.network/api
+    family: blockscout
+    name: Koi Explorer
+    url: https://koi-scan.darwinia.network/
+chainId: 701
```
Comment (jmrossy) [inline chains/koitestnet/metadata.yaml:7]: You have two chaind fields here, that's why the CI tests are failing

**Review by jmrossy — COMMENTED** [review]
Almost there

**Comment by guy93354** [comment]
Thanks Rossy ! Typo corrected,

**Comment by guy93354** [comment]
Everything is corrected, thank you.
